fdbserver/workloads/MemoryKeyValueStore.cpp

```diff
@@ -37,18 +37,20 @@ Key MemoryKeyValueStore::getKey(KeySelectorRef selector) const {
 	//Update the iterator position if necessary based on the value of orEqual
 	int count = 0;
 	if(selector.offset <= 0) {
-		if((selector.getKey() == mapItr->first && !selector.orEqual) || selector.getKey() != mapItr->first) {
+		if(mapItr == store.end() ||
+			 selector.getKey() != mapItr->first ||
+			 (selector.getKey() == mapItr->first && !selector.orEqual)) {
```

Comment:
Unrelated to your change, but your restructuring made me notice:

I think we can get rid of the check `selector.getKey() == mapItr->first` here. If we aren't equal to `mapItr->first`, then we would short-circuit in the 2nd clause. We could probably also reorder it so that checking orEqual happens before the key comparison for efficiency.

Reply:
Sure.  That also fixes whitespace.